merkle/map_verifier.go

```diff
@@ -0,0 +1,61 @@
+package merkle
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/storage"
+)
+
+// VerifyMapInclusionProof verifies that the passed in expectedRoot can be
+// reconstructed correctly given the other parameters.
+//
+// The process is essentially the same as the inclusion proof checking for
+// append-only logs, but with added support for "null" nodes as well as extra
```

Comment:
this last bit reads strangely and should probably be a separate sentence. You could even spoil the ending by revealing what the assumptions are if you like!